config/generic_update.py

```diff
@@ -0,0 +1,332 @@
+    def acquire_lock(self):
+        pass
+    def release_lock(self):
+        pass
+
+class PatchOrderer:
+    def order(patch):
+        # TODO: Implement patch orderer
+        pass
+
+class ChangeApplier:
+    def apply(change):
+        # TODO: Implement change applier
+        pass
+
+YANG_DIR = "/usr/local/yang-models"
+class ConfigWrapper:
+    def get_current_config(self):
+        config_db_json = __get_config_db_as_json()
+        return __convert_config_db_to_yang(config_db_json)
+
+    def validate_config(self, yang_as_json):
+        config_db_as_json = __convert_yang_to_config_db(yang_as_json)
+        config_as_yang
+
+        sy = sonic_yang.SonicYang(YANG_DIR, debug=debug)
+        sy.loadYangModel()
+        sy.loadData(config_db_as_json)
+
+        try:
+            sy.validate_data_tree()
+            return True
+        except Exception as ex:
+            return False
+
+    def get_config_db_as_json(self):
+        db_kwargs = dict(); data = dict()
+        configdb = ConfigDBConnector(**db_kwargs)
+        configdb.connect()
+        deep_update(data, FormatConverter.db_to_output(configdb.get_config()))
+        return FormatConverter.to_serialized(data)
+
+    def convert_config_db_to_yang(self, config_db_as_json):
+        sy = sonic_yang.SonicYang(YANG_DIR, debug=debug)
+        sy.loadYangModel()
+
+        yang_as_json = dict()
+        sy._xlateConfigDBtoYang(config_db_as_json, yang_as_json)
+
+        return yang_as_json
+
+    def convert_yang_to_config_db(self, yang_as_json):
+        sy = sonic_yang.SonicYang(YANG_DIR, debug=debug)
+        sy.loadYangModel()
+
+        config_db_as_json = dict()
+        sy.xlateJson = yang_as_json
+        sy.revXlateJson = config_db_as_json
+        sy._revXlateYangtoConfigDB(config_db_as_json, yang_as_json)
+
+        return yang_as_json
+
+class PatchApplier:
+    def __init__(self, patchorderer = PatchOrderer(), changeapplier = ChangeApplier(), configlocker = ConfigLocker(), configwrapper = ConfigWrapper()):
+        self.patchorderer = patchorderer
+        self.changeapplier = changeapplier
+        self.configlocker = configlocker
+
+    def apply(self, patch, format, verbose, dry_run):
+        if format == "ConfigDB":
+            yang_patch = __convert_config_db_patch_to_yang_patch(patch)
+        elif format == "YANG":
+            yang_patch = patch
+        else:
+            raise AttributeError(f"format argument value '{format}', the supported values are 'ConfigDB' or 'YANG'")
+
+        if verbose:
+            # TODO: Implement verbose logging
+            raise NotImplementedError
+
+        if dry_run:
+            # TODO: Implement dry-run
+            raise NotImplementedError
+
+        __acquire_config_lock()
+        old_config = __get_current_config()
+        target_config = __simulate_patch(patch, old_config)
+
+        if not(__validate(target_config)):
+            raise Exception(f"The given patch is not valid")
```

Comment:
>Exception [](start = 18, length = 9)

Not to throw general exception type. This looks a normal logic, not like a literal 'exception'. How about using return value?

Reply:
created custom exception `GenericConfigUpdaterError`